python/paddle/tensor/math.py

```diff
@@ -125,64 +127,76 @@
-
+        Tensor: A ``Tensor`` with type same as ``x``.
     Examples:
-
         .. code-block:: python
-
             import paddle
-            import paddle.fluid as fluid
-
-            x = fluid.data(name="x", shape=[32,32], dtype="float32")
-
-            # example 1: argument exponent is float
-            y_1 = paddle.pow(x, 2.0)
-            # y_1 is x^{2.0}
-
-            # example 2: argument exponent is Variable
-            exponent_tensor = fluid.layers.fill_constant([1], "float32", 3.0)
-            y_2 = paddle.pow(x, exponent_tensor)
-            # y_2 is x^{3.0}
+            import numpy as numpy
+            paddle.enable_imperative()
+            # example 1: exponent is a float
+            x_data = np.array([1, 2, 3])
+            exponent = 2
+            x = paddle.imperative.to_variable(x_data)
+            res = paddle.power(x, exponent)
+            print(res.numpy()) # [1 4 9]
+            
+            # example 2: exponent is a Variable
+            exponent = paddle.fill_constant(shape=[1], value=2, dtype='float32')
+            res = paddle.power(x, exponent)
+            print(res.numpy()) # [1 4 9]
     """
+    # in dynamic graph mode
     if in_dygraph_mode():
-        return core.ops.pow(input, "exponent", exponent)
-
-    helper = LayerHelper('pow', **locals())
-    inputs = {'X': input}
-    attrs = {}
-    if isinstance(exponent, Variable):
-        exponent.stop_gradient = True
-        inputs['FactorTensor'] = exponent
+        # exponent is scalar
+        if isinstance(exponent, (int, float)):
+            return core.ops.pow(x, 'factor', exponent)
+        # exponent is Tensor
+        else:
+            # if check_dtype(exponent.dtype, 'exponent', ['int32', 'int64']):
+            exponent = cast(exponent, dtype='float64')
+            x = cast(x, dtype='float64')
+            return _elementwise_op_in_dygraph(
+                x, exponent, axis=-1, act=None, op_name='elementwise_pow')
+    # in static graph mode
     else:
-        attrs['factor'] = exponent
+        # exponent is scalar
+        if isinstance(exponent, (int, float)):
+            helper = LayerHelper('pow', **locals())
+            inputs = {'X': x}
+            attrs = {'factor': exponent}
+            out = helper.create_variable_for_type_inference(dtype=x.dtype)
+            helper.append_op(
+                type='pow', inputs=inputs, outputs={'Out': out}, attrs=attrs)
+            return out
+        # exponent is Tensor
+        else:
+            #TODO elementwise_op c++ implementation only support single type calculation
+            # which will enhance in near future, here we cast different type by inserting 
+            # additional op cast
+            helper = LayerHelper('elementwise_pow', **locals())
+            y = exponent
+            if x[0].dtype != exponent[0].dtype:
+                exponent = cast(exponent, dtype='float64')
```

Comment:
why do you cast both exponent and y? (I think you cast twice)

Can you remove line 190 and line 191?